services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -284,4 +301,41 @@ public void run() {
     }
   }
 
+
+  private class CheckForNewPartitons implements Runnable {
+
+    public void run() {
```

Comment:
Some code seems redundant here. The logic for initializing should be in `start()` and the logic for clearing up state (e.g. close executor) should be in `close()`. The method should probably look something like this:

```
if (Utils.getPartitionNumForTopic(zkUrl, topic) > _partitionNum) {
  close();
  start();
}
```